File: Practica de 121/POO/5-estudiante/Estudiante.py
#Crea una clase Estudiante con nombre, nota_1, nota_2
class Estudiante:
    def __init__(self, nombre, nota1, nota2):
        self.nombre = nombre
        self.nota1 = nota1
        self.nota2 = nota2
    # a) Método para calcular el promedio
    def calcular_promedio(self):
        return (self.nota1 + self.nota2) / 2
    # b) Método para verificar si aprobó
    def aprobo(self):
        return self.calcular_promedio() >= 6
    def mostrar_resultado(self):
        return f"{self.nombre} - Promedio: {self.calcular_promedio():.1f} - Aprobó: {self.aprobo()}"
# c) Crear tres estudiantes y mostrar resultados
if __name__ == "__main__":
    estudiante1 = Estudiante("Juan", 7.5, 8.0)
    estudiante2 = Estudiante("María", 5.0, 6.5)
    estudiante3 = Estudiante("Carlos", 4.0, 3.5)

    print(estudiante1.mostrar_resultado())
    print(estudiante2.mostrar_resultado())
    print(estudiante3.mostrar_resultado())

File: Practica de 121/POO/5-estudiante/Estudiante.java
//Crea una clase Estudiante con nombre, nota_1, nota_2
public class Estudiante {
    private String nombre;
    private double nota1;
    private double nota2;

    public Estudiante(String nombre, double nota1, double nota2) {
        this.nombre = nombre;
        this.nota1 = nota1;
        this.nota2 = nota2;
    }

    // a Método para calcular el promedio
    public double calcularPromedio() {
        return (nota1 + nota2) / 2;
    }

    // b Método para verificar si aprobó
    public boolean aprobo() {
        return calcularPromedio() >= 6;
    }

    // c Crear tres estudiantes y mostrar resultados
    public static void main(String[] args) {
        Estudiante estudiante1 = new Estudiante("Juan", 7.5, 8.0);
        Estudiante estudiante2 = new Estudiante("María", 5.0, 6.5);
        Estudiante estudiante3 = new Estudiante("Carlos", 4.0, 3.5);

        System.out.println(estudiante1.nombre + " - Promedio: " + estudiante1.calcularPromedio() +
                " - Aprobó: " + estudiante1.aprobo());
        System.out.println(estudiante2.nombre + " - Promedio: " + estudiante2.calcularPromedio() +
                " - Aprobó: " + estudiante2.aprobo());
        System.out.println(estudiante3.nombre + " - Promedio: " + estudiante3.calcularPromedio() +
                " - Aprobó: " + estudiante3.aprobo());
    }
}